packages/calypso-products/src/types.ts

```diff
@@ -26,6 +26,25 @@ export type JetpackPlanSlugs =
 	| JetpackResetPlanSlugs
 	| JetpackLegacyPlanSlugs;
 
+export type JetpackProductSlug =
```

Comment:
This is not great since it requires to add new products manually. We could probably use arrays exported from `constants.js` to build the type, but at the moment it results in a generic string array. Transforming this file in a TS file in another PR might solve this.

Reply:
I think we can accomplish the same doing:

```
export type JetpackProductSlug = typeof JETPACK_PRODUCTS_LIST[number];
```

But we would have to add `as const` at the end of each arrays that compose `JETPACK_PRODUCTS_LIST` to tell TypeScript that those arrays won't change during runtime.

Reply:
Thanks Renzo. I'll keep note to try this in a later PR.